### ISSUE TITLE
Allow Entity Listener to match interface

### ISSUE THREAD

**Issue by hermann8u** [doctrine/orm#8117]
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no
| BC Break    | no

#### Summary

Hi,

There is a drawback using lifecycle listeners because they are called for every entities. On the other hand, entity listener can be applied only on one entity, limiting the number of calls, but not on interfaces.

Most of the time, when we use lifecycle listeners, the goal is to match an interface. That's why it will be great to have the possibility to use entity listeners mapped with an interface instead.

**Comment by greg0ire** [comment]
Possible next step: find the PR that introduced that feature, and see if this has already been discussed.

**Comment by hermann8u** [comment]
Yeah I tried to (maybe not enought?) to find isues and PR discuting about this. It's difficult to find something in more than 1100 issues and 250 PR though :/

So I guess there is something in that direction? Can I close this one?

**Comment by greg0ire** [comment]
> So I guess there is something in that direction?

Uuuh no? Since you did not find anything

>  Can I close this one?

Well, no, why would you close it?

> Yeah I tried to find isues and PR discuting about this (maybe not enought?). It's difficult to find something in more than 1100 issues and 250 PR though :/

It is, that's why you should use the git log instead.

**Comment by SimonVanacco** [comment]
I think this would be a huge QoL improvement to EntityListeners, is there any way to help achieve this ?

**Comment by derrabus** [comment]
> is there any way to help achieve this ?

Create a PR that implements this feature.

**Comment by janklan** [comment]
@SimonVanacco Implement a life cycle listener and ignore all events not related to the interface of your interest. That's what I did, anyway.
